### DISCUSSION TITLE
FSLeyes Closes Unexpectedly After Shimming-Toolbox Installation

### DISCUSSION THREAD

**Discussion by IgorTellez** [shimming-toolbox/shimming-toolbox#529] [Q&A] [answered]
I recently installed the shimming-toolbox following all instructions provided on the shimming-toolbox website. However, when I type shimming-toolbox in the terminal, the FSLeyes application starts for a few seconds and then automatically closes. Below are the errors shown on the terminal:

```
Tellez@MRTS5:~$ shimming-toolbox
Tellez@MRTS5:~$ WARNING:fsl.utils.idle:Idle task create crashed - wxAssertionError: C++ assertion ""tempContext"" failed at ./src/unix/glx11.cpp(498) in wxGLContext(): glXCreateContext failed
Traceback (most recent call last):
  File "/home/Tellez/shimming-toolbox/python/lib/python3.10/site-packages/fsl/utils/idle.py", line 576, in __idleLoop
    task.task(*task.args, **task.kwargs)
  File "/home/Tellez/shimming-toolbox/python/lib/python3.10/site-packages/fsleyes/gl/__init__.py", line 742, in create
    self.__createWXGLContext(requestVersion=requestVersion)
  File "/home/Tellez/shimming-toolbox/python/lib/python3.10/site-packages/fsleyes/gl/__init__.py", line 975, in __createWXGLContext
    ctx = wxgl.GLContext(self.__canvas, **candidate)
wx._core.wxAssertionError: C++ assertion ""tempContext"" failed at ./src/unix/glx11.cpp(498) in wxGLContext(): glXCreateContext failed
 WARNING              idle.py  578: __idleLoop      - Idle task create crashed - wxAssertionError: C++ assertion ""tempContext"" failed at ./src/unix/glx11.cpp(498) in wxGLContext(): glXCreateContext failed
Traceback (most recent call last):
  File "/home/Tellez/shimming-toolbox/python/lib/python3.10/site-packages/fsl/utils/idle.py", line 576, in __idleLoop
    task.task(*task.args, **task.kwargs)
  File "/home/Tellez/shimming-toolbox/python/lib/python3.10/site-packages/fsleyes/gl/__init__.py", line 742, in create
    self.__createWXGLContext(requestVersion=requestVersion)
  File "/home/Tellez/shimming-toolbox/python/lib/python3.10/site-packages/fsleyes/gl/__init__.py", line 975, in __createWXGLContext
    ctx = wxgl.GLContext(self.__canvas, **candidate)
wx._core.wxAssertionError: C++ assertion ""tempContext"" failed at ./src/unix/glx11.cpp(498) in wxGLContext(): glXCreateContext failed

```
**System Details:**
OS: Ubuntu 20.04.6 LT
FSL version: 6.0.7.11
FSLeyes version: 1.11.0
Shimming-toolbox version: v0.1

FSL and FSLeyes are already installed on the computer, and I can open FSLeyes without any issues independently.
I also installed the shimming-toolbox on a Macbook and works perfect.

**Request:**

Could you please provide support on this installation issue and suggest how I might resolve the problem? Any help would be greatly appreciated.

Thank you!

**Comment by po09i** [comment]
Hey @IgorTellez ! Thanks for the bug report, I'm looking into replicating your issue

**Reply by po09i** [reply, marked as answer]
@IgorTellez , I am having a hard time replicating your issue. However, at first glance, it seems to be related to the FSLeyes version and your Linus distro. Since you mentioned that you have FSLeyes installed and that it is working. We can try updating the FSLeyes version that Shimming Toolbox uses (it was pinned at 1.9.0). I updated to the latest release (1.12.4) and tested that it still works on MacOS.

To update to that version. Try the following:

```
git clone https://github.com/shimming-toolbox/fsleyes-plugin-shimming-toolbox.git
cd fsleyes-plugin-shimming-toolbox
make install CLEAN=true
# source your shell .XXXXrc file
source ~/.bashrc
shimming-toolbox
```
Let me know if you that fixes your issue

**Reply by IgorTellez** [reply]
Thank you very much, this solved the problem.
